chat-plugins/github.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * GitHub Alert Notifications
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * This notifies a set of rooms of changes to specific repositor(y/ies).
+ *
+ * Much of this comes from Ecuacion's bot, which is based on xfix's bot,
+ * and jd who originally converted this to PS server-side.
+ *
+ * @license MIT license
+ */
+
+'use strict';
+
+let gitConfig = {};
+
+if (!Config.github) {
+	return;
+} else if (Config.github === {}) {
+	gitConfig.port = 3420;
+	gitConfig.secret = "";
+} else {
+	gitConfig = {
+		port: Config.github.port,
+		secret: Config.github.secret,
+	};
+}
+
+const git = exports.github = require('githubhook')(gitConfig);
+
+let updates = {};
+let targetRooms = (Config.github.rooms && Config.github.rooms.length) ? Config.github.rooms : ['development'];
+targetRooms = targetRooms.map(toId).map(Rooms);
+let gitBans = {};
+if (targetRooms[0].chatRoomData) targetRooms[0].chatRoomData.gitBans = gitBans;
```

Comment:
This will overwrite `room.chatRoomData.gitBans` if the property's already present. Defining `gitBans` by using  `Object.assign` with each room's git bans, if present, might be a better idea